maintenance-editor/src/main/java/ca/on/oicr/gsi/shesmu/maintenance/Editor.java

```diff
@@ -0,0 +1,196 @@
+import javax.swing.table.TableCellRenderer;
+
+public class Editor extends JFrame {
+  private class DateTableModel extends AbstractTableModel {
+    private static final long serialVersionUID = 1L;
+    final List<LocalDateTime[]> rows = new ArrayList<>();
+
+    @Override
+    public Class<?> getColumnClass(int columnIndex) {
+      return LocalDateTime.class;
+    }
+
+    @Override
+    public int getColumnCount() {
+      return 2;
+    }
+
+    @Override
+    public String getColumnName(int column) {
+      return column == 0 ? "Start" : "End";
+    }
+
+    @Override
+    public int getRowCount() {
+      return rows.size();
+    }
+
+    @Override
+    public Object getValueAt(int row, int col) {
+      return rows.get(row)[col];
+    }
+
+    @Override
+    public boolean isCellEditable(int rowIndex, int columnIndex) {
+      return true;
+    }
+
+    @Override
+    public void setValueAt(Object aValue, int rowIndex, int columnIndex) {
+      rows.get(rowIndex)[columnIndex] = (LocalDateTime) aValue;
+    }
+  }
+
+  private static final long serialVersionUID = 1L;
+  private static final Pattern TAB = Pattern.compile("\t");
+
+  public static void main(String[] args) throws IOException {
+    File file;
+    if (args.length == 0) {
+      final JFileChooser chooser = new JFileChooser();
+      if (chooser.showOpenDialog(null) == JFileChooser.APPROVE_OPTION) {
+        file = chooser.getSelectedFile();
+      } else {
+        return;
+      }
+    } else if (args.length == 1) {
+      file = new File(args[0]);
+    } else {
+      System.err.println("Usage: maintenance-editor [maintenance.tsv]");
+      System.exit(1);
+      return;
+    }
+    new Editor(file).setVisible(true);
+  }
+
+  public Editor(File file) throws IOException {
+    setTitle(file.toString() + " - Maintenance Schedule Editor");
+    setSize(700, 400);
+    setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+    final JPanel panel = new JPanel(new BorderLayout());
+    add(panel);
+    final JToolBar toolbar = new JToolBar();
+    panel.add(toolbar, BorderLayout.PAGE_START);
+    final JButton add = new JButton("Add");
+    toolbar.add(add);
+    final JButton delete = new JButton("Delete");
+    toolbar.add(delete);
+    final JButton reload = new JButton("Revert");
+    toolbar.add(reload);
+    final JButton save = new JButton(UIManager.getIcon("FileView.floppyDriveIcon"));
+    toolbar.add(save);
+    save.setToolTipText("Save");
+    final DateTableModel model = new DateTableModel();
+    final JTable table =
+        new JTable(model) {
+          private static final long serialVersionUID = 1L;
+
+          public Component prepareRenderer(TableCellRenderer renderer, int row, int column) {
+            final JComponent c = (JComponent) super.prepareRenderer(renderer, row, column);
+            if (model.rows.get(row)[0].isAfter(model.rows.get(row)[1])) {
+              c.setBorder(BorderFactory.createMatteBorder(1, 1, 1, 1, Color.RED));
```

Comment:
Why a border? This might make sense if the table got sorted in ascending order but it doesn't seem to be sorted at the moment.

Reply:
The border appears if the window is invalid. That is, if start date is after end date.